File: csharp/test/Ice/inheritance/CCI.cs
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

namespace Ice
{
    namespace inheritance
    {
        public sealed class CCI : Test.MA.CCDisp_
        {
            public CCI()
            {
            }

            public override Test.MA.CAPrx caop(Test.MA.CAPrx p, Ice.Current current)
            {
                return p;
            }

            public override Test.MA.CCPrx ccop(Test.MA.CCPrx p, Ice.Current current)
            {
                return p;
            }

            public override Test.MB.CBPrx cbop(Test.MB.CBPrx p, Ice.Current current)
            {
                return p;
            }
        }
    }
}

File: java/test/Ice/inheritance/CCI.java
// **********************************************************************
//
// Copyright (c) 2002
// ZeroC, Inc.
// Billerica, MA, USA
//
// All Rights Reserved.
//
// Ice is free software; you can redistribute it and/or modify it under
// the terms of the GNU General Public License version 2 as published by
// the Free Software Foundation.
//
// **********************************************************************

public final class CCI extends MA.CC
{
    public
    CCI()
    {
    }

    public MA.CAPrx
    caop(MA.CAPrx p, Ice.Current current)
    {
        return p;
    }

    public MA.CCPrx
    ccop(MA.CCPrx p, Ice.Current current)
    {
        return p;
    }

    public MB.CBPrx
    cbop(MB.CBPrx p, Ice.Current current)
    {
        return p;
    }
}
